checker/jdk/nullness/src/java/util/Map.java

```diff
@@ -818,7 +818,7 @@ default V putIfAbsent(K key, V value) {
      *         (<a href="{@docRoot}/java/util/Collection.html#optional-restrictions">optional</a>)
      * @since 1.8
      */
-    default boolean remove(Object key, Object value) {
+    default boolean remove(@Nullable Object key, @Nullable Object value) {
```

Comment:
Similar `remove` methods, e.g. in `List` https://github.com/typetools/checker-framework/blob/master/checker/jdk/nullness/src/java/util/List.java#L278 are also annotated as `@Nullable Object`.
While a sound solution for this would be preferable, this isn't particular to this PR and can be solved separately.
I would suggest opening an issue to keep track of this soundness issue.

Reply:
These methods are particularly interesting because it's possible to have, say, a `HashMap<@Nullable String, @Nullable String>`. If the `remove` methods don't accept `null`, then there's no direct way to remove the null values that the API allows users to put in :) (Well, I suppose that they could retain a reference of static type `HashMap`, assuming that `HashMap.remove` is annotated to accept null values.)

As @wmdietl hints, even methods like [`Collection.contains`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html#contains(java.lang.Object)) and [`Map.get`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#get(java.lang.Object)) are allowed to reject `null`, which would prevent users from even _reading_ the values they put in.

All that said, you're certainly right that it's unsound. I had figured it was a bridge you've already crossed, but this may deserve more thought before going any further.

Perhaps what we "really" want is:

```java
boolean remove(@NullnessMatchesNullnessOf("K") key, @NullnessMatchesNullnessOf("V") value);
```

Reply:
I found the previous discussion of this: #1326

Reply:
Thanks for the link!

If you feel that you're in a hole here and you want to stop digging, I'm happy to revert. (I could even remove the existing `@Nullable` annotations if you'd like, though maybe that's more disruptive to existing users than you'd prefer?)

Reply:
How about putting the null-friendly annotations in a separate stub file?  See https://github.com/typetools/checker-framework/issues/3040.

Reply:
(That would require moving those changes to a different pull request.)

Reply:
Will do.